src/components/page/blocks/BlocksColumn.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Blocks from './index.jsx';
+
+function BlocksColumn({
+  getCol,
+  blocksGroups,
+  defaultLayout,
+  blockShared,
+  blockRows,
+}) {
+  const getLayout = layout => {
+    if (layout) {
+      const row = layout.row || defaultLayout.row;
+      const col = layout.col || defaultLayout.col;
+
+      return { row, col };
+    }
+
+    return defaultLayout;
+  };
+
+  const isInLayout = (layout, getRow) => {
+    const { row, col } = getLayout(layout);
+
+    return getCol === col && getRow === row;
+  };
+
+  const getRowBlockGroups = getRow =>
+    blocksGroups
+      .reduce((result, blockGroup) => {
+        if (blockGroup.blocks) {
+          result.push({
+            ...blockGroup,
+            blocks: blockGroup.blocks.filter(block => {
+              return isInLayout(block.layout, getRow);
+            }),
+          });
+        }
+
+        return result;
+      }, [])
+      .filter(blockGroup => blockGroup.blocks.length > 0);
+
+  return blockRows.map(getRow => {
```

Comment:
iterates through the rows (top, middle, and bottom as defaults) and extracts the block groups that match the current row and column then renders their blocks.